Kernel/Arch/init.cpp

```diff
@@ -417,6 +421,10 @@ void init_stage2(void*)
         PANIC("VirtualFileSystem::mount_root failed");
     }
 
+    // Initialise all USB Drivers
+    for (auto* init_function = driver_init_table_start; init_function != driver_init_table_end; init_function++)
```

Comment:
nitpick for a future PR: Wrap this in `USB::Initialize()` or some such and hide it in the USB directory.

Reply:
actually JK, nitpick for this PR, since CI is unhappy with your ternaries

```
/home/runner/work/serenity/serenity/Kernel/Bus/USB/USBManagement.cpp: In member function 'AK::LockRefPtr<Kernel::USB::Driver> Kernel::USB::USBManagement::get_driver_by_name(AK::StringView)':
/home/runner/work/serenity/serenity/Kernel/Bus/USB/USBManagement.cpp:87:24: error: operands to '?:' have different types 'std::nullptr_t' and 'AK::NonnullLockRefPtr<Kernel::USB::Driver>'
   87 |     return it.is_end() ? nullptr : *it;
      |            ~~~~~~~~~~~~^~~~~~~~~~~~~~~
  ```

Reply:
I think this mechanism can be reworked to work for all kinds of drivers, the type-name and comment just have to change

Reply:
> actually JK, nitpick for this PR, since CI is unhappy with your ternaries
> 
> ```
> /home/runner/work/serenity/serenity/Kernel/Bus/USB/USBManagement.cpp: In member function 'AK::LockRefPtr<Kernel::USB::Driver> Kernel::USB::USBManagement::get_driver_by_name(AK::StringView)':
> /home/runner/work/serenity/serenity/Kernel/Bus/USB/USBManagement.cpp:87:24: error: operands to '?:' have different types 'std::nullptr_t' and 'AK::NonnullLockRefPtr<Kernel::USB::Driver>'
>    87 |     return it.is_end() ? nullptr : *it;
>       |            ~~~~~~~~~~~~^~~~~~~~~~~~~~~
> ```

That should be fixed now

Reply:
sure.. but...
```
/home/runner/work/serenity/serenity/Kernel/Bus/USB/USBDevice.cpp: In static member function 'static AK::ErrorOr<AK::NonnullLockRefPtr<Kernel::USB::Device> > Kernel::USB::Device::try_create(const Kernel::USB::USBController&, u8, DeviceSpeed)':
/home/runner/work/serenity/serenity/Kernel/Bus/USB/USBDevice.cpp:38:14: error: variable 'result' set but not used [-Werror=unused-but-set-variable]
   38 |         auto result = driver->probe(device);
      |              ^~~~~~
  ```
  Perhaps building with GCC locally would be a good idea :sweat_smile:

Reply:
Huh wonder why clangd did not point out that one, but other unused warnings...
Should be fixed now

Reply:
Let's see if CI likes it this time